### PR TITLE
8265702: ZGC on macOS/aarch64

### PR DESCRIPTION
**Pull request by pliden**
This patch enables ZGC on macOS/aarch64. It does three things:
1) Enables building of ZGC on this platform.
2) Adds `os::processor_id()`, which for now always returns 0.
3) Fixes a WX issue in `OptoRuntime::handle_exception_C()`, where the stackwater mark might unnecessarily process a frame when the thread is in WXExec mode. In this case, the we're not touching any oops, so we don't need to process any frames.

Testing: Passes the same tests as macOS/x86_64 (with the exception of pre-existing issues unrelated to ZGC).

---------
### Progress
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue
- [x] Change must be properly reviewed

### Issue
 * [JDK-8265702](https://bugs.openjdk.java.net/browse/JDK-8265702): ZGC on macOS/aarch64

### Reviewers
 * [Erik Joelsson](https://openjdk.java.net/census#erikj) (@erikj79 - **Reviewer**) ⚠️ Review applies to c8913936aff026987c427f9a268def715b2eac0f
 * [David Holmes](https://openjdk.java.net/census#dholmes) (@dholmes-ora - **Reviewer**) ⚠️ Review applies to c8913936aff026987c427f9a268def715b2eac0f
 * [Stefan Karlsson](https://openjdk.java.net/census#stefank) (@stefank - **Reviewer**) ⚠️ Review applies to c8913936aff026987c427f9a268def715b2eac0f
 * [Gerard Ziemski](https://openjdk.java.net/census#gziemski) (@gerard-ziemski - Committer)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.java.net/jdk pull/3609/head:pull/3609` \
`$ git checkout pull/3609`

Update a local copy of the PR: \
`$ git checkout pull/3609` \
`$ git pull https://git.openjdk.java.net/jdk pull/3609/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 3609`

View PR using the GUI difftool: \
`$ git pr show -t 3609`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.java.net/jdk/pull/3609.diff">https://git.openjdk.java.net/jdk/pull/3609.diff</a>

</details>

### CODE REVIEW

**Review by erikj79 — APPROVED** [review]
Build change looks good.

File: src/hotspot/os/bsd/os_bsd.cpp [hunk 1]
```diff
@@ -2146,8 +2146,13 @@ uint os::processor_id() {
   assert(processor_id >= 0 && processor_id < os::processor_count(), "invalid processor id");
 
   return (uint)processor_id;
-}
+#else
```
Comment (dholmes-ora) [inline src/hotspot/os/bsd/os_bsd.cpp:2149]: Please add a comment indicating what platform this else is referring to.

**Review by dholmes-ora — APPROVED** [review]
Hi Per,

Looks good. One minor nit.

Thanks,
David

**Review by stefank — APPROVED** [review]

File: src/hotspot/os/bsd/os_bsd.cpp (hunk 1 above)
Comment (pliden) [reply src/hotspot/os/bsd/os_bsd.cpp:2149]: Will fix!

**Review by gerard-ziemski — APPROVED** [review]

**Comment by pliden** [comment]
Thanks @erikj79, @dholmes-ora, @stefank and @gerard-ziemski for reviewing!

/integrate

**Pull request closed without merging**